sympy/core/function.py

```diff
@@ -776,9 +775,11 @@ class UndefinedFunction(FunctionClass):
     """
     The (meta)class of undefined functions.
     """
-    def __new__(mcl, name, **kwargs):
-        ret = BasicMeta.__new__(mcl, name, (AppliedUndef,), kwargs)
-        ret.__module__ = None
+    def __new__(mcl, name, bases=(AppliedUndef,), __dict__=None, **kwargs):
+        __dict__ = __dict__ or {}
+        __dict__.update(kwargs)
```

Comment:
Is there a chance that this could mutate something passed by the caller? If so I think you could flip it around to be `kwargs.update(__dict__)` and pass `kwargs` instead.